modules/cli/src/main/scala/scala/cli/commands/Run.scala

```diff
@@ -100,8 +132,9 @@ object Run extends ScalaCommand[RunOptions] {
           .orExit(logger)
       builds.main match {
         case s: Build.Successful =>
-          maybeRun(s, allowTerminate = true)
+          val (process, _) = maybeRun(s, allowTerminate = true)
             .orExit(logger)
+          process.waitFor()
```

Comment:
Just like above, maybe we should wait for the on-exit future to finish (it's not clear whether it runs from a daemon thread or not… if that's a daemon thread, we have to wait for it, or it might not run if the main method exits before it runs).